plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/procedure/RegisterTableProcedure.java

```diff
@@ -0,0 +1,191 @@
+
+    @Inject
+    public RegisterTableProcedure(DeltaLakeMetadataFactory metadataFactory, TrinoFileSystemFactory fileSystemFactory, DeltaLakeConfig deltaLakeConfig)
+    {
+        this.metadataFactory = requireNonNull(metadataFactory, "metadataFactory is null");
+        this.fileSystemFactory = requireNonNull(fileSystemFactory, "fileSystemFactory is null");
+        this.registerTableProcedureEnabled = requireNonNull(deltaLakeConfig, "deltaLakeConfig is null").isRegisterTableProcedureEnabled();
+    }
+
+    @Override
+    public Procedure get()
+    {
+        return new Procedure(
+                SYSTEM_SCHEMA,
+                PROCEDURE_NAME,
+                List.of(
+                        new Procedure.Argument(SCHEMA_NAME, VARCHAR),
+                        new Procedure.Argument(TABLE_NAME, VARCHAR),
+                        new Procedure.Argument(TABLE_LOCATION, VARCHAR)),
+                REGISTER_TABLE.bindTo(this));
+    }
+
+    public void registerTable(
+            ConnectorSession clientSession,
+            String schemaName,
+            String tableName,
+            String tableLocation)
+    {
+        try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(getClass().getClassLoader())) {
+            doRegisterTable(
+                    clientSession,
+                    schemaName,
+                    tableName,
+                    tableLocation);
+        }
+    }
+
+    private void doRegisterTable(
+            ConnectorSession session,
+            String schemaName,
+            String tableName,
+            String tableLocation)
+    {
+        if (!registerTableProcedureEnabled) {
+            throw new TrinoException(PERMISSION_DENIED, "register_table procedure is disabled");
+        }
+        checkProcedureArgument(schemaName != null && !schemaName.isEmpty(), "schema_name cannot be null or empty");
+        checkProcedureArgument(tableName != null && !tableName.isEmpty(), "table_name cannot be null or empty");
+        checkProcedureArgument(tableLocation != null && !tableLocation.isEmpty(), "table_location cannot be null or empty");
+
+        SchemaTableName schemaTableName = new SchemaTableName(schemaName, tableName);
+        DeltaLakeMetastore metastore = metadataFactory.create(session.getIdentity()).getMetastore();
+
+        if (!metastore.getDatabase(schemaName).isPresent()) {
+            throw new TrinoException(SCHEMA_NOT_FOUND, format("Schema '%s' does not exist", schemaTableName.getSchemaName()));
+        }
+
+        if (metastore.getTable(schemaTableName.getSchemaName(), schemaTableName.getTableName()).isPresent()) {
+            throw new TrinoException(TABLE_ALREADY_EXISTS, format("Table '%s' already exists", schemaTableName));
+        }
+
+        TrinoFileSystem fileSystem = fileSystemFactory.create(session);
+        validateLocation(fileSystem, tableLocation);
+
+        Table.Builder tableBuilder = Table.builder()
```

Comment:
We should validate here that the location has a Delta table that we can actually read. Rather than just listing the _delta_log direction, you can do something like

```
            TableSnapshot tableSnapshot = transactionLogAccess.loadSnapshot(schemaTableName, new Path(tableLocation), session);
            if (transactionLogAccess.getMetadataEntry(tableSnapshot, session).isEmpty()) {
                throw ...;
            }
```

You'll need to inject a TransactionLogAccess instance

Reply:
I have added the validation as you suggested in `RegisterTableProcedure` to better handling and logging.

Reply:
Ah, sorry i missed that we had a `DeltaLakeMetastore`, not a `HiveMetastore`. I think it is fine to remove the location validations here and let the metastore checks run. 

Sorry for the back and forth